plugins/builds/index.js

```diff
@@ -293,83 +409,319 @@ exports.register = (server, options, next) => {
      * @param {Job}         config.job          Current job
      * @param {Build}       config.build        Current build
      * @param {String}      config.username     Username
-     * @param {String}      config.scmContext   scm context
+     * @param {String}      config.scmContext   Scm context
+     * @param {Boolean}     config.externalJoin Flag to allow external join
      * @return {Promise}                        Resolves to the newly created build or null
      */
-    server.expose('triggerNextJobs', (config) => {
-        const { pipeline, job, build, username, scmContext } = config;
+    server.expose('triggerNextJobs', async (config) => {
```

Comment:
This function is too large.
Could you separate this function into some functions?

Reply:
agree. it's bit hard to follow and this requires some refactoring too